### PR TITLE
Improving tests and stubs + createClient Stub

### PR DESCRIPTION
**Pull request by gauntface**
An awfully big PR - super sorry.

@wibblymat Would appreciate your input on this since its a few API changes to the push client. Either fixes to handle extremely rare errors in subscribe cases and the createClient factory method.

The rest of this is changes to the stub to account for these changes.

### CODE REVIEW

File: src/client/push-client.js [hunk 1]
```diff
@@ -255,10 +249,32 @@ export default class PushClient extends EventDispatch {
...
+    let options;
+    if (scope) {
+      options = {scope};
+    }
+    return navigator.serviceWorker.register(swPath, options)
+    .then(reg => {
+      return new PushClient(reg);
+    });
+  }
+
```
Comment (wibblymat) [inline src/client/push-client.js]: The docblock for isSupported has been accidentally removed here

File: src/client/push-client.js [hunk 2]
```diff
@@ -201,10 +195,10 @@ export default class PushClient extends EventDispatch {
    * Get the registration of the service worker being used for push.
    *
    * @return {Promise<ServiceWorkerRegistration>} A Promise that
-   *  resolves to either a ServiceWorkerRegistration or to null if none.
+   *  resolves to either a ServiceWorkerRegistration.
    */
   getRegistration() {
-    return this._registrationPromise;
+    return Promise.resolve(this._registration);
```
Comment (wibblymat) [inline src/client/push-client.js]: Perhaps just change the API to not return a Promise here? Seems unnecessary to have more async than we need.

**Comment by gauntface** [comment]
@wibblymat fixed both of those and fixed a few problems with travis.

**Comment by gauntface** [comment]
@wibblymat happy for this to land? Will enable me to land some of the service worker bits in other branches.

**Comment by wibblymat** [comment]
Ship it.

Sorry for not getting round to this!

**Pull request merged**